recipes/doxygen/all/conanfile.py

```diff
@@ -56,28 +41,34 @@ def requirements(self):
             self.requires("xapian-core/1.4.19")
             self.requires("zlib/[>=1.2.11 <2]")
         if self.options.enable_app or self.options.enable_parse:
-            # INFO: Doxygen uses upper case CMake variables to link/include IConv, so we are using patches for targets.
             self.requires("libiconv/1.17")
 
-    def compatibility(self):
-        return [{"settings": [("build_type", "Release")]}]
-
     def validate(self):
-        minimum_compiler_version = self._minimum_compiler_version.get(str(self.settings.compiler))
-        if minimum_compiler_version and Version(self.settings.compiler.version) < minimum_compiler_version:
-            raise ConanInvalidConfiguration(f"Compiler version too old. At least {minimum_compiler_version} is required.")
-        if Version(self.version) == "1.8.18":
-            check_min_vs(self, "191")
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "5":
+            raise ConanInvalidConfiguration("Doxygen requires GCC >=5")
+        
+        if self.settings.compiler == "msvc" and Version(self.settings.compiler.version) < "191":
+            raise ConanInvalidConfiguration("Doxygen requires Visual Studio 2017 or newer")
+        
+        check_min_cppstd(self, "17")
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows":
+        if self.settings_build.os == "Windows":
             self.tool_requires("winflexbison/2.5.24")
         else:
             self.tool_requires("flex/2.6.4")
             self.tool_requires("bison/3.8.2")
 
+        self.tool_requires("cmake/[>=3.19 <4]")
```

Comment:
This is to fix an issue seen on Windows  - newer versions of CMake have a more correct `find_package(Python)`